src/property.ts

```diff
@@ -1017,9 +1021,14 @@ function Property$setValue(property: Property, obj: Entity, currentValue: any, n
 	// Update lists as batch remove/add operations
 	if (property.isList) {
 		let currentArray = currentValue as ObservableArray<any>;
+
 		currentArray.batchUpdate((array) => {
 			updateArray(array, newValue);
 		}, additionalArgs);
+
+		// Set pendingInit to false here, since an array change event will not be raised if there are no changes
+		// (ex: array was previously an empty array and newValue is also an empty array).
+		Property$pendingInit(obj, property, false);
```

Comment:
Set pending init to false when a list property is set, regardless of whether it results in a change to the list